File: 3704.count-no-zero-pairs-that-sum-to-n.py
#
# @lc app=leetcode id=3704 lang=python3
#
# [3704] Count No-Zero Pairs That Sum to N
#

# @lc code=start
class Solution:
    def countNoZeroPairs(self, n: int) -> int:
        def is_no_zero(x):
            return '0' not in str(x)
        count = 0
        for a in range(1, n):
            b = n - a
            if is_no_zero(a) and is_no_zero(b):
                count += 1
        return count
# @lc code=end

File: 3704.count-no-zero-pairs-that-sum-to-n.cpp
#
# @lc app=leetcode id=3704 lang=cpp
#
# [3704] Count No-Zero Pairs That Sum to N
#
# @lc code=start
class Solution {
public:
    long long countNoZeroPairs(long long n) {
        auto isNoZero = [](long long x) {
            while (x > 0) {
                if (x % 10 == 0) return false;
                x /= 10;
            }
            return true;
        };
        long long count = 0;
        for (long long a = 1; a < n; ++a) {
            long long b = n - a;
            if (isNoZero(a) && isNoZero(b)) {
                ++count;
            }
        }
        return count;
    }
};
# @lc code=end